src/OpenTelemetry.Exporter.Jaeger/JaegerExporter.cs

```diff
@@ -158,6 +158,13 @@ internal void AppendSpan(JaegerSpan jaegerSpan)
             if (this.batchByteSize + spanTotalBytesNeeded >= this.maxPayloadSizeInBytes)
             {
                 this.SendCurrentBatch();
+
+                // SendCurrentBatch clears/invalidates the BufferWriter in InMemoryTransport.
+                // The new spanMessage is still located in it, though. It might get overwritten later
+                // when spans are written in the buffer for the next batch.
+                // Move spanMessage to the beginning of the BufferWriter to avoid data corruption.
+                this.memoryTransport.Write(spanMessage.BufferWriter.Buffer, spanMessage.Offset, spanMessage.Count);
```

Comment:
@luryus Possible to add a unit test for this? Something that would fail without the fix and then pass with it in.

Reply:
By the way, thanks for the fix sorry about creating this mess 😄

Reply:
No worries 😅

I couldn't figure out how this could be unit tested, at least without making more implementation details visible for the tests (I don't know if that would be acceptable or not?). In theory the serialised Batches could be validated in the tests, but the vendored Thrift library does not include read functionality so that's not easy either. But both opentelemetry and Thrift are completely new to me so I'd appreciate any suggestions.

Reply:
I added a test case that just appends multiple spans with pretty specific memory usage patterns, and validates that the last (unsent) batch contains the correct serialized spans. A bit shady but it works (fails without the bugfix, passes with it)

Reply:
Thanks @luryus! That's pretty much what I would have done 😄